### ISSUE TITLE
Feature request: filter or search by stack trace

### ISSUE THREAD

**Issue by barillax** [getsentry/sentry#1247]
A common annoying situation: a number of users are logging what appears to be the same error (and is bucketed as such), but only a small number have a browser that logs a useful stack trace. Usually means a lot of clicking in the hopes we stumble upon a useful one.

It would great if there was some way to filter and/or search by stack trace. Even something simple like hashing the first N lines of the stack trace so that its filterable like the rest of the tags would be a great time saver.

**Comment by howkey666** [comment]
+1

**Comment by filipmelik** [comment]
+1

**Comment by grongor** [comment]
+1

**Comment by mihu** [comment]
+1

**Comment by ygormutti** [comment]
+1

Not having an option to filter by stack trace has other consequences, besides the one described by the OP. When the same exception is thrown by multiple points in code it becomes very hard to find the entries related to a specific point. The same issue happens when you try to find exceptions that are not the last in the chain.

IMHO, this is the biggest flaw in Sentry today. The community could submit a PR to fix this, but this feature is so obvious that it seems it was left out on purpose. Can someone with knowledge about the codebase explain if this is the case?

Thanks!

**Comment by dcramer** [comment]
It's expensive to index every frame in the stack to make it searchable, and it's actually not that crippling without it. For example, if you have an exception happening in different parts of the stack, you can search on the exception (which is indexed).

We intend to do this in the future, but the cost tradeoffs will likely prevent this from happening for a little while yet.

**Comment by ygormutti** [comment]
Thanks for the explanation!

After you suggested searching on the exception I've realized

I have no idea what is indexed

Thanks, but how do I search on the exception? I'm having a hard time figuring out what search terms to use to find events caused by exceptions in Sentry.

For example, if I search for the name of the exception class (one I can see at the project initial page, so I'm sure it does exist) there are no results. None of the stack trace terms will do either. The same for the exception message.

Actually, I have no idea what IS indexed in exception events. The search feature is useless most of the time.

**Comment by ygormutti** [comment]
Thanks for the explanation, @dcramer!

After you suggested searching on the exception I've realized I have no idea what is indexed from exception events. I'm having a hard time figuring out what search terms to use to find events caused by exceptions.

For example, if I search for the name of the exception class (one I can see at the project initial page, so I'm sure it does exist) I get no results. None of the stack trace terms will do either. The same for terms in the exception message. The only results I get with the search feature are the ones that aren't related to exceptions. The problems of the search feature are worse than I have imagined.

Even when the results contain the exception I was looking for I often have to click through tens of events to find the one with the desired stack trace or additional data. From my experience, these issues make the search feature useless most of the time.
